test/regression/sql/materialized_view.sql

```diff
@@ -1,5 +1,5 @@
-CREATE TABLE t(a INT, b TEXT);
-INSERT INTO t SELECT g % 1000, MD5(g::TEXT) FROM generate_series(1,20000) g;
+CREATE TABLE t(a INT, b VARCHAR);
+INSERT INTO t SELECT g % 1000, MD5(g::VARCHAR) FROM generate_series(1,20000) g;
```

Comment:
The `SELECT` in the `CREATE MATERIALIZED VIEW` command now also goes through DuckDB. This is a good thing imho, because that means that we can use `read_csv` and `read_parquet` in those queries too.

However, on PG16 our type mapping results in the following error:
```
 CREATE MATERIALIZED VIEW tv AS SELECT * FROM t WHERE a % 100 = 0;
+ERROR:  SELECT rule's target entry 2 has different type from column "b"
+DETAIL:  SELECT target entry has type text, but column has type character varying.
```

Surprisingly in PG17 this works just fine.

While suboptimal, I think this is an improvement overall.